### PR TITLE
Autonomous Worlds on Substrate

### PR DESCRIPTION
**Pull request by driemworks**
## Project Abstract

**SpellCrafter** is a "spell-crafting" card game where players discover ingredients, unlock recipes, and craft new spells in a challenge to be the most powerful mage. Players encounter offchain challenges (puzzles) that must be solved in order to unlock new recipes that can be used to craft spells on-chain. This game is a re-imagining of [Arcane Assembly](https://github.com/ArcaneAssemblers/spellcrafter), a game developed with Dojo (on Starknet), as a Substrate-native game. Even though a game in it's own right, SpellCrafter is intended to be a used as a source of magic spells for other games that need them for their players. In the game, players must search the web, offchain, in order to learn knowledge that they can use to unlock recipes, such as by solving a puzzle or a riddle. 

To enable SpellCrafter on Substrate, we introduce publicly verifiable on-chain randomness, timelock encryption, verifiable computation (zkSNARKs) and practical witness encryption capabilities to web3 gaming on Polkadot/Substrate. In collaboration with ChainSafe Gaming, Ideal Labs aims to enhance web3 gaming on Polkadot by leveraging the Ideal Network's [post-finality gadget](https://medium.com/@ideal_labs/the-etf-post-finality-gadget-1dd6d7f12034), an MPC solution that enables a randomness beacon, to develop a practical witness encryption mechanism. That is, it enables the usage of verifiable randomness on-chain (and across chains), allows for messages to be locked for future blocks, and provides secret sharing capabilities, wherein the network acts as a distributed key generator and constructs keys for 'authorized' users (where authorization is gated behind valid proofs). This approach ensures fair play, enables asynchronous player coordination, and introduces proof-of-knowledge gating for in-game assets, events, or anything. This would not only allow for games like [zkHunt](https://0xparc.org/blog/zk-hunt) or [dark forest](https://zkga.me/) to be implemented on Substrate, but would allow for new on-chain conditional data access mechanisms to be created, such as requiring a zkSNARK of some in-game knowledge in order to proceed to the next level.

## Application Checklist

[x] The [application template](https://github.com/PolkadotPlay/GamesBounty/blob/main/applications/application-template.md) or [delivery template](https://github.com/PolkadotPlay/GamesBounty/blob/main/deliveries/delivery-template.md) has been copied and aptly renamed for an application (`project_name.md`) and a delivery (`project_name_m1.md`).
[x] I have read the [games bounty guidelines](https://github.com/PolkadotPlay/GamesBounty/blob/main/docs/gamesbounty_guidelines.md).
[ ] Payment details have been provided (Polkadot address in the application). (will add shortly)
[x] The initial PR contains only one commit (squash and force-push if needed).

### CODE REVIEW

**Comment by darkfriend77** [comment]
Hey @driemworks 

Thank you for the application, is this ready for an initial review?

File: applications/autonomous_worlds_on_substrate.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
+# SpellCrafter
+###### Autonomous Worlds and Conditional Access Control for Substrate Native Games
+
+- **Team Name:** Ideal Labs and Chainsafe Systems
+- **Funding Details:**
+  - **DOT**: For the **DOT** compensation, please provide a Polkadot address. (ex. multisig, with core contributors)
```
Comment (juangirini) [inline applications/autonomous_worlds_on_substrate.md:6]: We'll provide this address later

**Comment by juangirini** [comment]
> Hey @driemworks
> 
> Thank you for the application, is this ready for an initial review?

@darkfriend77 yes, this is ready for review

**Comment by boorich** [comment]
🤞🏼 ...

**Comment by carloskiron** [comment]
💪

**Comment by HopeClary** [comment]
Dear Application Team,

I hope this message finds you well. I would like to provide some feedback on the game’s bounty application:

Game Design Document (GDD):
The current GDD does not seem to reflect aspects related to the spelling and the powerful mage game loop. It still appears to be centered around the concept of "Fragments." Additionally, there are no visuals provided for the spell narrative. And how you will be distinct from the Arcane Assembly Game (https://github.com/ArcaneAssemblers/spellcrafter)

Similar Games:
In the section discussing similar games, Xaya and Mud are mentioned. These are game engine frameworks/projects rather than actual games. Please provide examples of playable games that are similar to the proposed project.

Interaction Time and Time lock Randomness:
Clarification is needed on the interaction time for time lock randomness and execution. Specifically, how long does a player need to wait for these interactions? What is the minimum time required for such interactions?

Randomness Mechanism:
Instead of using a very expensive randomness function for this, wouldn't a traditional coin flip randomness mechanism be sufficient for this game? Considering the typical value of game assets, a simpler method might be much more practical and more efficient. High-stakes games, such as those in the casino and gambling sectors, would benefit probably more from tech-intensive solutions.

GDD and Spell Crafting Details:
There seems to be a focus on the spells being usable in other games, but what is the incentive for another game to use spells from SpellCrafter, as opposed to minting and selling their own NFTs that they have full control of? Additionally, how easy would it be for you to use the spell system in another game? Will there be any APIs/libraries/plugins/etc. that simplify the interaction with the system, or will game developers be expected to implement blockchain interaction themselves?

The lack of a comprehensive GDD detailing the spell-crafting process makes it difficult to understand the core mechanics of the game. It appears that the game is designed more around a technical use case rather than utilizing technology to enhance a game feature. 

Bounty and Grant Structure:
It may be more appropriate to split the bounty into a traditional Web3 Foundation grant for Milestones 1 and 2, as they are very technical. As a suggestion only Milestone 3 should be included in the game's bounty, accompanied by a proper GDD.

Thank you for considering this feedback. I look forward to seeing the improvements and clarifications in the revised application.

Best regards,

The Games Bounty Team

**Comment by carloskiron** [comment]
@HopeClary thanks a lot for this helpful feedback. We are currently putting in place some changes based on your detailed comments. I will ping you back once we have it ready for a new review. Thanks again.

**Comment by darkfriend77** [comment]
**Subject: Important Update on Games Bounty – Transition to DOT Play Bounty**

Dear Team,

We hope this message finds you well. We are writing to inform you of an important update regarding the Games Bounty you expressed interest in. 

As you may be aware, the Games Bounty has recently undergone a significant transformation and has now been rebranded as the DOT Play Bounty. As part of this transition, the curators of the original Games Bounty no longer have access to the associated funds, nor do we have any authority over the allocation or spending of the 100k DOT that was initially set aside for this initiative.

We would like to take this opportunity to sincerely thank you for your interest and the time you have invested in the Games Bounty. Your enthusiasm and dedication to contributing to the Polkadot ecosystem are greatly appreciated.

While the Games Bounty has been discontinued in its previous form, we encourage you to explore the following avenues to continue your involvement:

1. **Contact DOT Play:** If you are interested in participating in the newly formed DOT Play Bounty, we recommend you contact the DOT Play team for further details and potential opportunities.

2. **Create a Treasury Proposal:** Consider submitting a proposal through the Polkadot governance system. This is an excellent way to secure funding for innovative projects that align with the broader Polkadot ecosystem.

3. **Web3 Foundation Grants:** You may also explore opportunities through the Web3 Foundation by applying for a grant. The foundation continues to support promising projects that contribute to the growth and development of the Web3 ecosystem.

We are confident that your talents and ideas will find the right platform for success, and we encourage you to continue your journey within the Polkadot and Web3 communities.

Thank you once again for your understanding and continued contributions.

Best regards,  
Cedric

Former Games Bounty Curator

**Pull request closed without merging**